include/tenstorrent/uart_tt_virt.h

```diff
@@ -185,7 +185,7 @@ static inline int tt_vuart_poll_in(volatile struct tt_vuart *vuart, unsigned cha
 
 		if (atomic_compare_exchange_strong(headp, &head, head + 1)) {
 			*p_char = vuart->buf[offs + (head % cap)];
-			return *p_char;
```

Comment:
this change seems to break the api

> @return The incoming byte, or -1 (EOF) if the buffer is empty

Reply:
Fixed, I'll just check the return value in `uart_tt_virt_poll_in`